freqtrade/strategy/interface.py

```diff
@@ -424,14 +424,18 @@ def gather_informative_pairs(self) -> ListPairsWithTimeframes:
         informative_pairs = self.informative_pairs()
```

Comment:
This will fail - informative_pairs is a user-exposed interface function - so we have to assume that we get 2 item tuples back, not one with 3.

having compatibility code for that should be pretty simple (but will be necessary), as that 3rd argument is really not necessary.

Reply:
We'll probably save futures candles as `XRP/USDT:USDT`

Reply:
I want to force 3 values until just before this PR is merged, it helps me catch errrors

Reply:
> I want to force 3 values until just before this PR is merged, it helps me catch errrors

it really won't as you've updated the tests to go along with it.
So now the tests are passing (no longer testing a 2 value tuple) - but the error will be there once an existing strategy is ran against this code.

Reply:
I guess so, it's just helped me so far

Reply:
@xmatthias I don't know how to make the third string optional. I tried these types

- `PairWithTimeframe = Union[Tuple[str, str], Tuple[str, str, str]]`
- `PairWithTimeframe = Tuple[str, str, Optional(str)]`

but just receive errors like

`Argument 1 to "refresh_latest_ohlcv" of "Exchange" has incompatible type "List[Tuple[str, str, str]]"; expected "List[Union[Tuple[str, str], Tuple[str, str, str]]]"`

Reply:
Let me be honest - i think this is details on a PR that's greatly lacking overall, which can be left "till the end".
I think the fact that there is no description in the issue explains a bit how this PR is going.

I'm not demanding/asking for a description on a pr that's clear (where it's clear what it aims to provide), in this case however, it feels like you know we'll need this data (mark, index and funding rates), but have little idea / understanding on how this should be done.
That's not per se a problem - and this understanding can for sure be gained, or i can help with that. The way it is however is that you've just started "somewhere", and made 100ds of changes that may or may not be necessary in random places throughout the code - and are now losing yourself in details (such as the above) - which may not even be relevant for the end-result. 


I'm still not seeing much of an idea on the following topics:
* how will we make sure to always load the right data (reliably, across multiple exchanges with quite different naming standards). Right now, you're assuming that pairs are always different - which we cannot rely on - AND which is wrong if we look at binance, too. CCXT may change this on binance - but i understand it'll be non-breaking - which means BTC/USDT will remain a valid futures pair for the time being - which means we'll have to be sure we can reliably determine that it's a different pair.  
  We'll ALSO need a way to reliably filter pairs based on the data available - which is needed for several commands, so we'll need a reliable way from the filename/location to determine if the containing data is futures data, spot data, or mark/index candles.
* How can the user select which data to download? (user interface)   
  Maybe they'll simply specify `download-data <...>  futures`- and freqtrade will then download all data that's necessary for backtesting futures?   
  Alternatively, we'll have the user manually download all required data (which can be cumbersome and is most likely problematic)

---

I'm proposing the same again (what i already proposed somewhere here).
I'll take this PR from here (just give it time ... ) - you focus on "the other" PR's, where it's clearer what the methods you're changing are actually doing, and i'll take this one from here - as i honestly don't see this moving in the direction i'd like it to. Hope this works for you.

Reply:
> CCXT may change this on binance - but i understand it'll be non-breaking - which means BTC/USDT will remain a valid futures pair for the time being - which means we'll have to be sure we can reliably determine that it's a different pair.

I opened a [PR on CCXT](https://github.com/ccxt/ccxt/pull/10720) which I think will more or less solve the issues with Binance futures haven't separate symbols, how it works is that symbols that look like `ADA/USDT` or `ADA/USDT:USDT` but when CCXT see's the `ADA/USDT` symbols when the defaultType is `future` or `delivery`, then it will convert the symbol to `ADA/USDT:USDT`. So it will be able to read the `ADA/USDT` symbols, but will always return symbols that look like `ADA/USDT:USDT` and this will probably continue for the near future, until we completely switch to `ADA/USDT:USDT` symbols and then the `ADA/USDT` symbols will no longer be accepted

Reply:
yes, and this is how i understood it.
but this "intermediate" period is the problem. 
I cannot rely on a user to have the pairlist with `ADA/USDT:USDT` if the `ADA/USDT` is also valid.

No matter what ccxt will return - both pairs will be listed in the `markets` dictionary - which makes both correct.

Reply:
> 
The markets dictionary would only have the new pairs, the structure accepting a defaultType as `spot`, `margin`, `future` and `delivery` will still be in there in the intermediate period, so if you pass a symbol like `ADA/USDT` but your default type is `future`, then it grabs the symbol `ADA/USDT:USDT` instead and works with that, the user should never see the `ADA/USDT` symbols when using future or delivery unless they code those symbols themselves, 

The PR hasn't been merged, so something could possibly change, but this is the intent so far